### ISSUE TITLE
Raspberry pi install fail

### ISSUE THREAD

**Issue by WR1171** [YunoHost-Apps/pleroma_ynh#81]
Greetings!

Looking back on #45 & #57  and I don't think its the same issue I have.

I have one instance on Digital Ocean with Yunohost + Pleroma, running solid.  Loving the install, it was a breeze.

Now installed Yunohost on my Raspberry pi 3 b+ - looking to install Pleroma, but run into fails.  It doesn't even look like I am getting to MIX with issues.  Brain is boggling, tried may reflash/versions, installs etc still no luck.

2019-06-03 00:21:59,522: DEBUG - + ynh_die '--message=Unable to install dependencies'
2019-06-03 00:21:59,523: DEBUG - + local legacy_args=mc
2019-06-03 00:21:59,524: DEBUG - + args_array=([m]=message= [c]=ret_code=)
2019-06-03 00:21:59,524: DEBUG - + declare -Ar args_array
2019-06-03 00:21:59,525: DEBUG - + local message
2019-06-03 00:21:59,526: WARNING - Unable to install dependencies
2019-06-03 00:21:59,528: DEBUG - + local ret_code
2019-06-03 00:21:59,528: WARNING - !!
2019-06-03 00:21:59,530: DEBUG - + ynh_handle_getopts_args '--message=Unable to install dependencies'
2019-06-03 00:21:59,531: WARNING -   pleroma's script has encountered an error. Its execution was cancelled.
2019-06-03 00:21:59,533: DEBUG - + set +x
2019-06-03 00:21:59,535: WARNING - !!
2019-06-03 00:21:59,537: DEBUG - + echo 'Unable to install dependencies'
2019-06-03 00:21:59,538: DEBUG - + exit 1
2019-06-03 00:21:59,539: DEBUG - + ynh_exit_properly
2019-06-03 00:21:59,539: DEBUG - + local exit_code=1
2019-06-03 00:21:59,540: DEBUG - + '[' 1 -eq 0 ']'
2019-06-03 00:21:59,541: DEBUG - + trap '' EXIT
2019-06-03 00:21:59,542: DEBUG - + set +eu
2019-06-03 00:21:59,543: DEBUG - + echo -e '!!\n  pleroma'\''s script has encountered an error. Its execution was cancelled.\n!!'
2019-06-03 00:21:59,543: DEBUG - + type -t ynh_clean_setup
2019-06-03 00:21:59,544: DEBUG - + ynh_clean_setup
2019-06-03 00:21:59,545: WARNING - 
2019-06-03 00:21:59,547: DEBUG - + ynh_clean_check_starting
2019-06-03 00:21:59,548: DEBUG - + kill -s 15
2019-06-03 00:21:59,548: DEBUG - kill: usage: kill [-s sigspec | -n signum | -sigspec] pid | jobspec ... or kill -l [sigspec]
2019-06-03 00:21:59,549: DEBUG - + ynh_secure_remove ''
2019-06-03 00:21:59,550: DEBUG - + local legacy_args=f
2019-06-03 00:21:59,551: DEBUG - + args_array=([f]=file=)
2019-06-03 00:21:59,552: DEBUG - + declare -Ar args_array
2019-06-03 00:21:59,553: DEBUG - + local file
2019-06-03 00:21:59,553: DEBUG - + ynh_handle_getopts_args ''
2019-06-03 00:21:59,554: DEBUG - + set +x
2019-06-03 00:21:59,555: DEBUG - + echo '! Helper used in legacy mode !'
2019-06-03 00:21:59,555: DEBUG - + set +x
2019-06-03 00:21:59,556: DEBUG - + local 'forbidden_path= 	/var/www 	/home/yunohost.app'
2019-06-03 00:21:59,556: DEBUG - + '[' 1 -ge 2 ']'
2019-06-03 00:21:59,557: DEBUG - + [[  	/var/www 	/home/yunohost.app =~ '' ]]
2019-06-03 00:21:59,557: DEBUG - + echo 'Avoid deleting .'
2019-06-03 00:21:59,558: DEBUG - Avoid deleting .
2019-06-03 00:21:59,558: DEBUG - + ynh_die
2019-06-03 00:21:59,559: DEBUG - + local legacy_args=mc
2019-06-03 00:21:59,559: DEBUG - + args_array=([m]=message= [c]=ret_code=)
2019-06-03 00:21:59,560: DEBUG - + declare -Ar args_array
2019-06-03 00:21:59,560: DEBUG - + local message
2019-06-03 00:21:59,560: DEBUG - + local ret_code
2019-06-03 00:21:59,561: DEBUG - + ynh_handle_getopts_args
2019-06-03 00:21:59,561: DEBUG - + set +x
2019-06-03 00:21:59,562: DEBUG - + echo ''
2019-06-03 00:21:59,562: DEBUG - + exit 1

**Comment by yalh76** [comment]
Hi @WR1171 

Can you launch the install of pleroma from the command line using `yunohost app install https://github.com/YunoHost-Apps/pleroma_ynh --debug` and give the link to the paste.yunohost.org logs :)

**Comment by WR1171** [comment]
Greetings!

Here it is:

https://paste.yunohost.org/raw/cuqakuwada

Thanks for the help, I have been pulling my hair out, tried multiple installs, different images, fresh installs, manual installs, can't figure it out.  DigitalOcean bootups were super easy, I am having trouble with this arm business!!

**Comment by yalh76** [comment]
ok seems an issue installing dependencies 'elixir erlang-dev erlang-tools erlang-parsetools erlang-eldap erlang-xmerl'

What other yunohost package have you installed on you raspberry pi ?

**Comment by WR1171** [comment]
I haven't installed anything.  I used the image from yuonohost for raspberry pi - ran the commands, created domain/user, community apps, followed by installing pleroma.  I keep getting the error,

**Comment by yalh76** [comment]
Can you try to execute:
`echo "deb https://packages.erlang-solutions.com/debian stretch contrib" | sudo tee /etc/apt/sources.list.d/erlang-solutions.list`

`wget https://packages.erlang-solutions.com/debian/erlang_solutions.asc`

`sudo apt-key add erlang_solutions.asc`

`sudo apt update`

`sudo apt-get install -s elixir erlang-dev erlang-tools erlang-parsetools erlang-eldap erlang-xmerl`

and paste the result of the last command

**Comment by WR1171** [comment]
Here it is!



Reading package lists... Done
Building dependency tree
Reading state information... Done
Some packages could not be installed.  This may mean that you have requested an impossible situation or if you are using the unstable distribution that some required packages have not yet been created or been moved out of Incoming.
The following information may help to resolve the situation:

The following packages have unmet dependencies:
 elixir : Depends: erlang-base (> = 1:20.3) but 1:20.1.7-1 is to be installed or
                                           erlang-base-hipe (>= 1:20.3) or
                                           esl-erlang (>= 1:20.3) but it is not going to be installed
           Depends: erlang-crypto (>= 1:20.3) or
                          esl-erlang (>= 1:20.3) but it is not going to be installed
           Depends: erlang-inets (>= 1:20.3) or
                          esl-erlang (>= 1:20.3) but it is not going to be installed
           Depends: erlang-ssl  (>= 1:20.3) or
                          esl-erlang (>= 1:20.3) but it is not going to be installed
           Depends: erlang-syntax-tools (>= 1:20.3) or
                          esl-erlang (>= 1:20.3) but it is not going to be installed
E: Unable to correct problems, you have held broken packages.

**Comment by yalh76** [comment]
Seems that Elixir request erlang-base version 1.20.3 or above and only version 1.20.1.7 is available for raspberry pi, 
You will have to wait for a new release of erlang-base

**Comment by WR1171** [comment]
Thank you for following up on this, I have been pulling my hair out.  I have been pretty impressed with yunohost and the ease of installing some great apps.

After booting up the raspberry pi I tried so many different routes with no success.

I will look into this and see if I can keep things moving forward!

Cheers

**Comment by WR1171** [comment]
Thanks @yalh76 for following up on my issues, I have finally figured it out (sort of?), is it possible to fork this install/app and create one just for raspberry pi? asdf can be used to install erlang & elixir, if it can be put into the plugin, everything else can be installed normally. would be great to have a yunohost app to install via raspberry pi!  I would do it, but I have no idea what I am doing!

**Comment by yalh76** [comment]
What version of Raspberry Pi do you have ?

**Comment by WR1171** [comment]
Tried with both Raspberry pi 2, 3+ b, tinkerboard and Rockpro64 - all using Dietpi

**Comment by yalh76** [comment]
Can you try to reinstall the Raspberry Pi 3+ with the YunoHost image: https://yunohost.org/#/install_on_raspberry_en and then try to install pleroma ?

**Comment by WR1171** [comment]
Yup, I am pretty sure I actually did that a few times before I started using Dietpi.  I will have to wait until I am home before, is there a new image for RP?

I ran into this without using Yunohost at all as well, and getting almost the same error.  But after using ASDF to install elixir/erlang I was able to get it working.  Went down a few rabbit holes to get it working on a domain vs local host, which is why I live Yunohost so much, will try this and update you when I get back to my house to try it out.

THanks for following up!
